### PR TITLE
User Management Enhancements

### PR DESCRIPTION
**Pull request by drunkleen**
User Management Enhancements

### Description:
This pull request introduces several important enhancements to the user management functionality. The changes made include the addition of new methods, fixes to existing methods, and updates to improve user interactions.

### Changes Made:

- [x] Search Users Method Added
- [x] Like and Unlike Methods Fixed
- [x] Follow and Unfollow Methods Updated
- [x] Block and Unblock Methods Added
- [x] Mute and Unmute Methods Added
- [x] Restrict and Unrestrict Methods Added
- [x] Check Friendship Status with a User
- [x] README.md updated

If there are any concerns, feedback, or suggestions, I am more than willing to address them promptly.
Thank you for your time and consideration.

### CODE REVIEW

File: threadspy/threads_api.py [hunk 1]
```diff
@@ -487,39 +488,278 @@ def __toggle_auth__post_request(self, url: str):
...
+            boolean and if verbose mode is enabled, prints response dict
+        """
+        user_id = self.user_id or self.get_current_user_id()
+        response = self.__toggle_auth__post_request(
+            url=f"{BASE_API_URL}/api/v1/media/{post_id}_{user_id}/like/",
         )
-        return res.json()["status"] == "ok"
+        if self.verbose:
+            print("[LIKE]", response.json())
+        return response.json()["status"] == "ok"
```
Comment (junhoyeo) [inline threadspy/threads_api.py]:
```suggestion
        data = response.json()
        if self.verbose:
            print("[LIKE]", data)
        return data["status"] == "ok"
```
Can we store the `response.json()` into another variable? (Same for other usages, too)

File: README.md [hunk 2]
```diff
@@ -137,17 +185,24 @@ api.unfollow(user_id_to_follow)
...
-  - [x] ✅ Create new Thread with text
-    - [x] ✅ Make link previews to get shown
-  - [x] ✅ Create new Thread with media
+- [ ] 🚧 Write data (i.e. write automated Threads)
+  - [ ] 🚧 Create new Thread with text
+    - [ ] 🚧 Make link previews to get shown
+  - [ ] 🚧 Create new Thread with media
   - [ ] 🚧 Create new Thread with multiple images
-  - [x] ✅ Reply to existing Thread
+  - [ ] 🚧 Reply to existing Thread
```
Comment (junhoyeo) [inline README.md:193]: Thought we had `api.publish` — correct me if I'm wrong!

**Review by junhoyeo — APPROVED** [review]
Whoa, that was fast! Thanks :)

File: threadspy/threads_api.py (hunk 1 above)
Comment (drunkleen) [reply threadspy/threads_api.py]: Sure, I guess we should change it in any other methods as well

File: README.md (hunk 2 above)
Comment (drunkleen) [reply README.md:193]: Yeah but they changed the URL, and every time I tried its a 404 status

File: threadspy/threads_api.py (hunk 1 above)
Comment (junhoyeo) [reply threadspy/threads_api.py]: yup

File: README.md (hunk 2 above)
Comment (junhoyeo) [reply README.md:193]: @drunkleen oh I see, I'll merge this as-is and test/fix in some other PR. 🚀

File: threadspy/threads_api.py (hunk 1 above)
Comment (junhoyeo) [reply threadspy/threads_api.py]: can you change them?

Comment (drunkleen) [reply threadspy/threads_api.py]: I'm going to cover that in my next pull request, as well as the issue with the publishing URL. =)

Comment (junhoyeo) [reply threadspy/threads_api.py]: omg thanks 🙏

Comment (drunkleen) [reply threadspy/threads_api.py]: No need to thank me, I'm using the library too xD

**Pull request merged**